### PR TITLE
CI: Add cmake build and test

### PR DESCRIPTION
**Pull request by andistorm**
Since https://github.com/EVerest/everest-framework/pull/118 is kind of stale, I tried to do my best to hook it up.
What I did:

* rebased on main
* Adapted things to current build-kit image
* Split build and test in two steps
* Embed unit tests in CTest

@dorezyuk Thank you for your work here! Please feel to comment here

@a-w50, regarding your concerns:
* Lets keep using the build-kit image here for now, we can refactor/optimize this later
* Installing rust in the workflow isn't ideally but ok for now I would say

I would like to get this merged to have this check on PRs when working on everest-framework.
Furthermore the unit tests are working again so we should use them..

### CODE REVIEW

File: .ci/build-kit/compile.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+
+# Install Rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+source "$HOME/.cargo/env"
```
Comment (andistorm) [inline .ci/build-kit/compile.sh]:
```suggestion
# Skip the Rust setup until we can properly support Rust.
# Install Rust
# curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
# source "$HOME/.cargo/env"
```

File: .ci/build-kit/compile.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+cmake \
+    -B build \
+    -S "$EXT_MOUNT/source" \
+    -G Ninja \
+    -DEVC_ENABLE_CCACHE=1 \
+    -DBUILD_TESTING=ON \
+    -DEVEREST_ENABLE_RS_SUPPORT=ON
```
Comment (andistorm) [inline .ci/build-kit/compile.sh]:
```suggestion
    -DEVEREST_ENABLE_RS_SUPPORT=OFF # Turn off rust support until we can properly support Rust.
```

File: .ci/build-kit/test.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -e
+
+cd build
+ninja test
```
Comment (andistorm) [inline .ci/build-kit/test.sh]:
```suggestion
ninja -j$(nproc) -C build test
```

File: .github/workflows/lint.yaml
Comment (andistorm) [inline .github/workflows/lint.yaml:1]: don't delete

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -1,3 +1,4 @@
+include(CTest)
```
Comment (andistorm) [inline tests/CMakeLists.txt]:
```suggestion
```

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**